File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String regExp = "[^a-zA-Z0-9]";
        s = s.replaceAll(regExp, "").toLowerCase();
        for (int i = 0; i < s.length() / 2; i++) {
          if (s.charAt(i) != s.charAt(s.length() - 1 - i)) {
            return false;
          }
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        new_string = [char.lower() for char in s if char.isalnum()]
        
        for i in range(len(new_string) // 2):
            if new_string[i] != new_string[len(new_string) - 1 - i]:
                return False

        return True